src/messages/MOSDOp.h

```diff
@@ -375,6 +404,7 @@ struct ceph_osd_request_head {
       encode(flags, payload);
       encode(reqid, payload);
       encode_trace(payload, features);
+      encode_otel_trace(payload, features);
```

Comment:
Yeah, I still don't understand why we have both.  Can we simply remove the old one?  Is it actually possible to use both at the same time?

Reply:
there was some additional work in https://github.com/ceph/ceph/pull/48971 that removes all of the blkin tracing stuff, but that was delayed behind this PR. it would be convenient to take that for squid too, but that branch is over a year out of date now

it doesn't look like any of the librados interfaces let you pass in both kinds of traces to be encoded at the same time, but the config option `osd_blkin_trace_all=true` could enable the blkin trace on message decode even if the sender didn't encode one. so while it is _possible_ for both tracing systems to be in use at the same time, i don't think that's useful

Reply:
@cbodley Ok, if the plan is to actually remove blkin than I'm fine with it.  I *really* want to avoid accumulating tracing mechanics that need to be passed through our public librados api.

Reply:
> I _really_ want to avoid accumulating tracing mechanics that need to be passed through our public librados api.

full agreement there 👍

we haven't formally discussed deprecation/removal of the blkin tracing stuff. we should at least announce its deprecation for squid

Reply:
@rzarzynski Feelings on announcing blkin as deprecated for squid?

Reply:
More than fine with that!

Reply:
thanks, i've opened https://github.com/ceph/ceph/pull/55499 for this

Reply:
The deprecation PR is merged. I think we can resolve this thread.